tests/phpunit/Framework/ClassNameTest.php

```diff
@@ -33,44 +33,82 @@
 
 declare(strict_types=1);
 
-namespace Infection\Tests\AutoReview;
+namespace Infection\Tests\Framework;
 
-use Infection\Testing\SourceTestClassNameScheme;
+use Closure;
+use Infection\Framework\ClassName;
 use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\DataProvider;
 use PHPUnit\Framework\TestCase;
 
-#[CoversClass(SourceTestClassNameScheme::class)]
-final class SourceTestClassNameSchemeTest extends TestCase
+#[CoversClass(ClassName::class)]
+final class ClassNameTest extends TestCase
 {
+    /**
+     * @param class-string $className
+     */
+    #[DataProvider('classNameProvider')]
+    public function test_it_can_get_a_class_short_name(
+        string $className,
+        string $expectedShortName,
+    ): void {
+        $actual = ClassName::getShortClassName($className);
+
+        $this->assertSame($expectedShortName, $actual);
+    }
+
+    public static function classNameProvider(): iterable
+    {
+        yield 'nominal' => [
+            ClassName::class,
+            'ClassName',
+        ];
+
+        yield 'UTF-8' => [
+            'Webmozarts\ClassName\Ütf8',
+            'Ütf8',
+        ];
+
+        yield 'emoji' => [
+            'Webmozarts\ClassName\Special😋Class',
```

Comment:
technically picked it from the libraries of my workplace which are under a proprietary license. It's fairly boring code, so no need for license approval here. Left the name as a nod.